src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataFunctionalTest.java

```diff
@@ -0,0 +1,243 @@
+package uk.gov.hmcts.reform.cwrdapi;
+
+import io.restassured.response.Response;
+import lombok.extern.slf4j.Slf4j;
+import net.thucydides.core.annotations.WithTag;
+import net.thucydides.core.annotations.WithTags;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.cwrdapi.controllers.request.CaseWorkerRoleRequest;
+import uk.gov.hmcts.reform.cwrdapi.controllers.request.CaseWorkersProfileCreationRequest;
+import uk.gov.hmcts.reform.cwrdapi.controllers.response.SearchStaffUserResponse;
+import uk.gov.hmcts.reform.cwrdapi.controllers.response.StaffWorkerSkillResponse;
+import uk.gov.hmcts.reform.cwrdapi.util.FeatureToggleConditionExtension;
+import uk.gov.hmcts.reform.cwrdapi.util.ToggleEnable;
+import uk.gov.hmcts.reform.lib.util.serenity5.SerenityTest;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.cwrdapi.util.FeatureToggleConditionExtension.getToggledOffMessage;
+
+@ComponentScan("uk.gov.hmcts.reform.cwrdapi")
+@WithTags({@WithTag("testType:Functional")})
+@ActiveProfiles("functional")
+@SerenityTest
+@SpringBootTest
+@Slf4j
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class StaffRefDataFunctionalTest extends AuthorizationFunctionalTest {
+    public static final String STAFF_REF_DATA_SKILL_URL = "/refdata/case-worker/skill";
+    public static final String STAFF_REF_DATA_RD_STAFF_UI_KEY =
+            "StaffRefDataController.retrieveAllServiceSkills";
+    public static final String SEARCH_STAFF_USER_BY_NAME_KEY = "StaffRefDataController.searchStaffUserByName";
+    public static final String CASE_WORKER_PROFILE_URL = "/refdata/case-worker/profile";
+
+    @Test
+    @ToggleEnable(mapKey = STAFF_REF_DATA_RD_STAFF_UI_KEY, withFeature = true)
+    @ExtendWith(FeatureToggleConditionExtension.class)
+    public void should_return_service_skills_with_status_code_200_when_flag_true() {
```

Comment:
These skill related tests are already covered in https://github.com/hmcts/rd-caseworker-ref-api/pull/584/files#diff-8d51055287307302fe906e24c0670cc56d9851e86439519b44fb30eb05703f83R40 . May I know why you have included it again ?

Reply:
removed skills related duplicate test cases